src/FacetedTokenInput.js

```diff
@@ -214,6 +231,14 @@ export default class FacetedTokenInput extends Component {
       else {
         this.refs.facetedTokenInput.focus();
       }
+
+      // this will detect when the input is empty and allow the text
+      // to change direction if the prop is not defined
+      if (!dir && !dirChange && tokens.length === 0 && searchText === '') {
+        this.setState({
+          dirChange: true
+        });
+      }
```

Comment:
If the input is totally empty (no tokens and no text) the direction can change again. 

This isn't the best place to check if the input is empty since this function is on a loop (even if it's on a loop setState is never called more than 1 time) but it prevents at least one small error.

If the detection is placed inside the onChange or the onBackspace function:

* `onChange`: if the last token in the input is deleted the function doen't get called even if the input is empty

* `onBackspace`: if the last token is deleted the tokens.lenght doesn't get updated immediately leaving the direction locked even if the input is empty.